### PR TITLE
WFCORE-1130 Add support for an ObjectMapAttributeDefinition

### PR DESCRIPTION
**Pull request by ctomc**

### CODE REVIEW

**Comment by ctomc** [comment]
@darranl here you go.

File: controller/src/main/java/org/jboss/as/controller/ObjectMapAttributeDefinition.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            super(name);
+            this.valueType = valueType;
+            setElementValidator(valueType.getValidator());
+            setAttributeParser(AttributeParsers.OBJECT_MAP_WRAPPED);
+            setAttributeMarshaller(AttributeMarshaller.OBJECT_MAP_MARSHALLER);
+        }
+
+        public static ObjectListAttributeDefinition.Builder of(final String name, final ObjectTypeAttributeDefinition valueType) {
+            return new ObjectListAttributeDefinition.Builder(name, valueType);
+        }
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/ObjectMapAttributeDefinition.java]: This should return ObjectMapAttributeDefinition.Builder, not ObjectList...

**Comment by bstansberry** [comment]
@darranl IIRC you were the driver for this enhancement. Does this look like what you wanted?

Some test links that give you a sense:

https://github.com/wildfly/wildfly-core/pull/1575/files#diff-dadbae2ed567e9d636a3418915dc447dR243
https://github.com/wildfly/wildfly-core/pull/1575/files#diff-bd0359fe391dd52f43fa6e39fc89727dR4

The minor thing I noted above, this seems ok to me.

**Comment by darranl** [comment]
+1 that does look like what we were looking for.

File: controller/src/main/java/org/jboss/as/controller/ObjectMapAttributeDefinition.java (hunk 1 above)
Comment (ctomc) [inline controller/src/main/java/org/jboss/as/controller/ObjectMapAttributeDefinition.java]: ups :) fixed.

**Comment by ctomc** [comment]
@bstansberry @darranl do you guys maybe want this also for 2.x branch. If so i can prepare PR.

**Comment by darranl** [comment]
All my work is on 3 only so not for me thank you.

**Comment by bstansberry** [comment]
@ctomc Let's stick to 3.x.

**Pull request merged**